src/main/webapp/query-control.js

```diff
@@ -290,6 +290,52 @@ function replaceValueIfEmpty(dataListId) {
   }
 }
 
+// When the user picks certain dropdown values,
+// exclude the combinations that don't have data available
+const dropdownsToExclude = new Map();
+dropdownsToExclude.set(
+      'worked in', ['children'],
+    ).set(
+      'moved to', ['children', 'adults'],
+    ).set(
+      'children', ['worked in', 'moved to'],
+    ).set(
+      'adults', ['moved to'],
+    );
+const defaultDropdowns = new Map();
+defaultDropdowns.set(
+      'action',
+      [['lived in', 'live'], ['worked in', 'work'], ['moved to', 'moved']],
+    ).set(
+      'person-type',
+      [['people', 'all-ages'], ['adults', 'over-18'], ['children', 'under-18'],
+      ['men', 'male'], ['women', 'female']],
+    );
+
+function updateDropdown(type) {
+  // When we have an input in one area,
+  // we actually want to update the other area based on this input
+  let dropdown;
+  let options;
+  if (type === 'action') {
+    dropdown = document.getElementById('person-type');
+    options = defaultDropdowns.get('person-type');
+  } else {
+    dropdown = document.getElementById('action');
+    options = defaultDropdowns.get('action');
+  }
+  dropdown.innerHTML = '';
+  const currentValue = document.getElementById(type + '-list').value;
+
+  options.forEach((option) => {
+    if (!(dropdownsToExclude.has(currentValue) &&
```

Comment:
I think a comment explaining this if statement would be helpful

Reply:
Sure, good point! Added.